src/test/ui/consts/const-eval/promoted_errors.rs

```diff
@@ -8,21 +8,34 @@
 
 #![warn(const_err, arithmetic_overflow, unconditional_panic)]
 
+// The only way to have promoteds that fail is in `const fn` called from `const`/`static`.
+const fn overflow() -> u32 {
+    0 - 1 //~WARN arithmetic_overflow
+    //[opt_with_overflow_checks,noopt]~^ WARN any use of this value will cause an error
+}
+const fn div_by_zero1() -> i32 {
+    1 / 0 //[opt]~WARN unconditional_panic
+    //[opt]~^ WARN any use of this value will cause an error
+}
+const fn div_by_zero2() -> i32 {
+    1 / (1-1)
+}
+const fn div_by_zero3() -> i32 {
+    1 / (false as i32)
+}
+const fn oob() -> i32 {
+    [1,2,3][4]
+}
+
+const X: () = {
+    let _x: &'static u32 = &overflow();
+    //[opt_with_overflow_checks,noopt]~^ WARN any use of this value will cause an error
+    let _x: &'static i32 = &div_by_zero1();
+    //[opt]~^ WARN any use of this value will cause an error
```

Comment:
Curiously, the errors stop after the first panic. This should change if we decide that promoteds get added to (or treated like) `required_consts`.

Reply:
hmm... interesting. We must have an early abort somewhere in const prop

Reply:
No, ConstProp is not involved I think. The error is emitted by ConstEval. And of course ConstEval stops on the first error.